### DISCUSSION TITLE
Process to verify if Business Object Exist - Do we use a Checker Object now?

### DISCUSSION THREAD

**Discussion by jhuerta3** [MarimerLLC/csla#4255] [Questions] [answered]
Hello all,
I have a question regarding the Exist function. Before we could create a business object(Info or edit) and call the Exist Command to see if the data exist in the database. I think that method is no longer used since the business object does not expose the Exist Command. I believe we now should use a Checker object(ReadOnlyBase). Is that the case? If so why is this a better method? Just curious. 
Thank you.

**Comment by Chicagoan2016** [comment, marked as answer]
I have used command objects for 'Exists' pattern in Csla 4.xx, but I have also  used another method in my objects that works perfectly fine with our SQL server database(s). 
When I 'fetch' an object, I check for the property that corresponds to the Primary property (identity column) of the (main) table in the SQL server. If it's greater than zero, I know the fetch was successful. If not, I show the appropriate message to the user.
Our objects aren't 1-1 with SQL tables but we always have a 'main' table that corresponds to the object graph.

**Reply by jhuerta3** [reply]
Thank you @Chicagoan2016

**Reply by Chicagoan2016** [reply]
Glad to help
